crd/overlayextensionconfig/api/v1alpha1/overlayextensionconfig_types.go

```diff
@@ -0,0 +1,57 @@
+//go:build !ignore_uncovered
+// +build !ignore_uncovered
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OverlayExtensionConfig is the Schema for the overlayextensionconfigs API
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:shortName=oec
+// +kubebuilder:printcolumn:name="OverlayExtensionConfig IP range",type=string,priority=1,JSONPath=`.spec.extensionIPRange`
+type OverlayExtensionConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OverlayExtensionConfigSpec   `json:"spec,omitempty"`
+	Status OverlayExtensionConfigStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// OverlayExtensionConfigList contains a list of OverlayExtensionConfig
+type OverlayExtensionConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OverlayExtensionConfig `json:"items"`
+}
+
+// OverlayExtensionConfigSpec defines the desired state of OverlayExtensionConfig
+type OverlayExtensionConfigSpec struct {
+	// +kubebuilder:validation:Optional
+	ExtensionIPRange string `json:"extensionIPRange,omitempty"`
+}
+
+type OECState string
+
+const (
+	OECNone      OECState = "OECNone"
+	OECPending   OECState = "OECPending"
+	OECSucceeded OECState = "OECSucceeded"
+	OECFailed    OECState = "OECFailed"
+)
+
+// OverlayExtensionConfigStatus defines the observed state of OverlayExtensionConfig
+type OverlayExtensionConfigStatus struct {
+	// +kubebuilder:validation:Enum=OECNone;OECPending;OECSucceeded;OECFailed
+	State   OECState `json:"state,omitempty"`
+	Message string   `json:"message,omitempty"`
```

Comment:
events is another way tot od this as you can only leave one message here but fine for now.